### PR TITLE
improve README.md for redis local installation

### PR DESCRIPTION
**Pull request by adityabisoi**
This PR closes issue #38 

@rkpattnaik780 @ankitjena @coderwassananmol  please review this and suggest changes, if required.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,29 +1,47 @@
...
+#### Without Docker
+
+* Refer to [Redis](https://redis.io/download) for download and setup documentation
+
+#### Using Docker
+
+* `docker run --name redis -p 6379:6379 redis` and `docker start redis` or
+* **docker-compose** : `docker-compose up -d` and `docker-compose start`
+
+### Installing
```
Comment (rkpattnaik780) [inline README.md]: Too many bullet points under this heading, can you divide it into some appropriate sub-headings? 2 - 3 subheadings will suffice hopefully.

File: README.md [hunk 2]
```diff
@@ -1,29 +1,47 @@
...
+  * Go to https://archive.org/account/s3.php . Generate the **access** and **secret** keys and fill 
+     them in the `.env` file in the corresponding fields.
+  * Go to [Google Developers console](https://console.developers.google.com/getting-started). Make a new project to run the app. In that Google Developers project, search for 'Books API' in the Google API console, then **enable** the API for the project, then generate the **API keys**, and then copy and paste the API key in the `GB_Key` field.
+  * Fill the `redishost` field with **docker**  if you are using docker, otherwise **127.0.0.1**
+  * Fill the `redisport` field with **6379**, which is the default port number for redis.
+  * Fill `service` field with **gmail**.
+Note: In order to send email through gmail, you may need to allow less secure app access. To turn it on, go to https://myaccount.google.com/lesssecureapps?pli=1
+* Run `npm run dev` for development and `npm run start` for production.
+* Open your browser and run your local server: http://localhost:5000
```
Comment (rkpattnaik780) [inline README.md]: Open your browser and navigate to [http://localhost:5000](http://localhost:5000)

File: README.md [hunk 3]
```diff
@@ -1,29 +1,47 @@
...
+
+## Contributing
+
+Please read [CONTRIBUTING.md](https://github.com/coderwassananmol/BUB2/blob/develop/CONTRIBUTING.md) for the process for submitting pull requests to us.
```
Comment (rkpattnaik780) [inline README.md]: Make it "for information on how to contribute to BUB2."

**Comment by adityabisoi** [comment]
I have made the necessary changes. Thanks.

File: README.md [hunk 4]
```diff
@@ -1,29 +1,50 @@
...
 
+These instructions will get you a copy of the project up and running on your local machine for development and testing purposes. See deployment for notes on how to deploy the project on a live system.
+
+### Prerequisites
+
+* [Node.JS](https://nodejs.org/en/download/) ( check for installation with ```npm -v``` and ```node -v``` on terminal )
+* [Docker toolbox](https://docs.docker.com/toolbox/toolbox_install_windows/) ( Optional )
+* [Redis](https://redis.io/)
+
+#### Without Docker
```
Comment (rkpattnaik780) [inline README.md]: Redis setup should be an instruction within `Local Setup`. Place it before `Start the server`. And it doesn't need to have two sub-headings.

File: README.md [hunk 5]
```diff
@@ -1,29 +1,50 @@
...
+
+* Refer to [Redis](https://redis.io/download) for download and setup documentation
+
+#### Using Docker
+
+* `docker run --name redis -p 6379:6379 redis` and `docker start redis` or
+* **docker-compose** : `docker-compose up -d` and `docker-compose start`
+
+### Installing
```
Comment (rkpattnaik780) [inline README.md]: Change it to `Local Setup`.

File: README.md [hunk 6]
```diff
@@ -1,29 +1,50 @@
...
+#### Clone the repo
+
+* Clone the repository `git clone https://github.com/coderwassananmol/BUB2`
+* Navigate to the project directory on the terminal: `cd BUB2`.
+* For developers, checkout to develop branch: `git checkout develop`
+* Run `npm install`
+#### Enter credentials
```
Comment (rkpattnaik780) [inline README.md]: Make it `Set Environment Variables`.

File: README.md (hunk 4 above)
Comment (adityabisoi) [reply README.md]: @rkpattnaik780 do you want me to remove with & without docker subheadings?

Comment (rkpattnaik780) [reply README.md]: Yeah, inside the local setup, just do something like this : -

### Run Redis Server 
* Refer to [Redis](https://redis.io/download) for download and setup documentation

#### Running Redis using Docker
    docker run --name redis -p 6379:6379 redis
    docker start redis

#### Running Redis using Docker Compose
    docker-compose up -d
    docker-compose start

Comment (adityabisoi) [reply README.md]: I have done the necessary changes. Please review. Thanks.

**Pull request merged**